### PR TITLE
feat: [PM4-95] sorting and filter for tables

### PR DESCRIPTION
**Pull request by SandroGuerotto**
PM4-95
Added sorting for Tables: my-reservations, reservations of my parkinglots, admin-users, admin-parkinglots-
added filter for reservations of my parkinglots.

I deliberately did not add any sorting or more filtering for the search-table

### CODE REVIEW

File: frontend/src/components/search-parking-lot/search-parking-lot-table.tsx [hunk 1]
```diff
@@ -19,13 +24,43 @@ const SearchParkingLotTable = ({
     'reservieren'
   ];
 
+  const bookNowLink = (searchItem: SearchResultModel) => (
+    <Link
+      href="#"
```
Comment (DaRabus) [inline frontend/src/components/search-parking-lot/search-parking-lot-table.tsx]: das funktioniert?

ben grad uswendig ned secher, dued dass ned site neu lade?

Comment (DaRabus) [reply frontend/src/components/search-parking-lot/search-parking-lot-table.tsx]: Sost machs eifach zo nem "div" ond hau det de "onClick" druf, typography sett au goh, esch aber je nach font chli komesch

File: frontend/src/components/reservation/confirmation-modal.tsx [hunk 2]
```diff
@@ -57,7 +59,17 @@ const ConfirmationModal = ({
   const [toDate, setToDate] = useState<Date>(data.toDate);
 
   const validateInput = (): boolean => {
-    return fromDate <= toDate && fromDate >= new Date() && toDate >= new Date();
+    logger.log({ fromDate, toDate });
```
Comment (DaRabus) [inline frontend/src/components/reservation/confirmation-modal.tsx]: Can remove that I guess

File: frontend/src/components/my-reservation/reservation-cancel-cell.tsx [hunk 3]
```diff
@@ -18,10 +18,10 @@ const CancelCell = ({
   reservation,
   onClick
 }: CancelCellProps): string | JSX.Element => {
-  if (new Date(reservation.from) <= minCancelDate()) {
-    return <span></span>;
-  } else if (reservation.cancelDate) {
+  if (reservation.cancelDate) {
     return `${formatDate(new Date(reservation.cancelDate))}`;
+  } else if (new Date(reservation.from) <= minCancelDate()) {
+    return <span></span>;
```
Comment (DaRabus) [inline frontend/src/components/my-reservation/reservation-cancel-cell.tsx]: empty span?

chash sost au </> mache oder "null"

File: frontend/src/components/my-parking-lot-reservations/parking-lots-filter.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      justifyContent="space-between"
+      display="flex"
+      spacing={1}
+      rowSpacing={2}
+      padding={2}
+    >
+      <Grid item md={5} lg={5} xl={5} sm={8}>
+        <TextField
+          label={'Suchbegriff'}
+          fullWidth={true}
```
Comment (DaRabus) [inline frontend/src/components/my-parking-lot-reservations/parking-lots-filter.tsx]:
```suggestion
          fullWidth
```
true state doesn't have to be defined

**Review by DaRabus — APPROVED** [review]
Datatable, we probably couldn't use cause of Mui premium?

Kinda hacky but should be fine for that usecase we are using it, and since our data array.

Maybe fix the small comments i made :)

File: frontend/src/components/my-reservation/reservation-cancel-cell.tsx (hunk 3 above)
Comment (SandroGuerotto) [reply frontend/src/components/my-reservation/reservation-cancel-cell.tsx]: weiss au nüme wieso span. evtl will am ahfang nöd richtig grendert het mit null.. hans uf </> gändert

File: frontend/src/components/search-parking-lot/search-parking-lot-table.tsx (hunk 1 above)
Comment (SandroGuerotto) [reply frontend/src/components/search-parking-lot/search-parking-lot-table.tsx]: han uf div gwechselt und eigens style gmacht

**Pull request merged**